integration-tests/tests/src/tests/notifications.ts

```diff
@@ -55,6 +58,87 @@ describe("Notifications", () => {
 
   beforeEach(() => (runCount = 0));
 
+  /**
+   * Helper function to use as a generic handler for ordered collection listeners.
+   * @param onEnd Callback to run once the listener reaches the end of the changes array. If the callback is Mocha.Done, it will ensure that
+   * the listener was run exactly [changesOnRun.length] times
+   * @param changesOnRun An array of Realm.CollectionChangeSet.
+   * The handler will assert equality of listener's changes with the next element of the array on every run.
+   */
+  const expectCollectionChangesOnEveryRun = (
+    onEnd: () => void | Mocha.Done,
+    changesOnRun: Realm.CollectionChangeSet[],
+  ) => {
+    return (_: Realm.OrderedCollection<any>, changes: Realm.CollectionChangeSet) => {
+      runCount++;
+
+      expect(changes.insertions).deep.equals(changesOnRun[runCount - 1].insertions);
+      expect(changes.oldModifications).deep.equals(changesOnRun[runCount - 1].oldModifications);
+      expect(changes.newModifications).deep.equals(changesOnRun[runCount - 1].newModifications);
+      expect(changes.deletions).deep.equals(changesOnRun[runCount - 1].deletions);
+
+      if (runCount >= changesOnRun.length) {
+        // Once runCount reaches given array, run the onEnd callback. If onEnd is Mocha.done then
+        // running it multiple times will cause tests to fail so this also ensures the listener fires
+        // exactly changesOnRun.length times.
+        onEnd();
+      }
+    };
+  };
+
+  /**
+   * Helper function to use as a generic handler for dictionary listeners.
+   * @param onEnd Callback to run once the listener reaches the end of the changes array. If the callback is Mocha.Done, it will ensure that
+   * the listener was run exactly [changesOnRun.length] times
+   * @param changesOnRun An array of Realm.DictionaryChangeSet.
+   * The handler will assert equality of listener's changes with the next element of the array on every run.
+   */
+  const expectDictionaryChangesOnEveryRun = (
+    onEnd: () => void | Mocha.Done,
+    changesOnRun: Realm.DictionaryChangeSet[],
+  ) => {
+    return (_: Realm.Dictionary<any>, changes: Realm.DictionaryChangeSet) => {
+      runCount++;
+
+      expect(changes.insertions).members(changesOnRun[runCount - 1].insertions);
```

Comment:
I would probably store this `changesOnRun[runCount - 1]` in its own `const` before using it three times.

Reply:
Ditto in the collection helper above.